Add plugin: Dynamic Editor Width

bwya77 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/bwya77/dynamic-editor-width

## Release Checklist
- [X] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
Please do not include the generated main.js file in your repo.

[if (this.settings.editorWidth === this.settings.editorWidthDefault) {](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L39-L42)
This code seems unnecessary. If it is needed, can you explain why?

[.setName("Default Editor Width")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L266), [.setName("Show Width Control in Status Bar")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L287), [.setName("Control Style")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L298), [.addOption("input", "Input Box")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L301)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[new Setting(containerEl)](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L313-L315)
This note should be closer to the related setting. I recommend updating the description of the "Default editor width" setting to include this additional information instead of having it's own setting at the bottom of the list of settings.

---

I know you don't have it in your releases, but consider removing the styles.css file from your repo as well if you aren't using it.

bwya77 commented:
> Please do not include the generated main.js file in your repo.
> 
> [if (this.settings.editorWidth === this.settings.editorWidthDefault) {](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L39-L42) This code seems unnecessary. If it is needed, can you explain why?
> 
> [.setName("Default Editor Width")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L266), [.setName("Show Width Control in Status Bar")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L287), [.setName("Control Style")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L298), [.addOption("input", "Input Box")](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L301) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> [new Setting(containerEl)](https://github.com/bwya77/dynamic-editor-width/blob/0d34423aa041f49067716145fd224e4e32d4826f/main.ts#L313-L315) This note should be closer to the related setting. I recommend updating the description of the "Default editor width" setting to include this additional information instead of having it's own setting at the bottom of the list of settings.
> 
> I know you don't have it in your releases, but consider removing the styles.css file from your repo as well if you aren't using it.

Modified .gitignore to not include styles.css
Removed the block of code that did nothing
edited the settings UI to use sentence case
Integrated the YAML frontmatter description into the default editor width setting description

bwya77 commented:
and updated the release files to include new main.js

joethei commented:
What sets your plugin apart from the "Editor Width Slider" / "Custom note width" plugins?

As your plugin is based on the "Editor Width Slider" plugin, why did you not contribute your changes to that plugin?

[inputContainer.style.cssText = `](https://github.com/bwya77/dynamic-editor-width/blob/84f589ac544919dbcab2523184cd54b5c0d30c9b/main.ts#L58-L63), [input.style.cssText = `](https://github.com/bwya77/dynamic-editor-width/blob/84f589ac544919dbcab2523184cd54b5c0d30c9b/main.ts#L80-L89), [valueDisplay.style.cssText = `](https://github.com/bwya77/dynamic-editor-width/blob/84f589ac544919dbcab2523184cd54b5c0d30c9b/main.ts#L113-L119), [slider.style.cssText = `](https://github.com/bwya77/dynamic-editor-width/blob/84f589ac544919dbcab2523184cd54b5c0d30c9b/main.ts#L129-L132)
This should be in the `styles.css` file, and not set via JS.

bwya77 commented:
Im just going to close this
